JSMenu.cpp

```diff
@@ -52,14 +45,16 @@ JSAPI_FUNC(my_selectChar) {
     if (argc != 1 || !JSVAL_IS_STRING(JS_ARGV(cx, vp)[0]))
         THROW_ERROR(cx, "Invalid parameters specified to selectCharacter");
 
-    char* profile = JS_EncodeString(cx, JS_ValueToString(cx, JS_ARGV(cx, vp)[0]));
+    const wchar_t* profile = JS_GetStringCharsZ(cx, JS_ValueToString(cx, JS_ARGV(cx, vp)[0]));
+
     if (!Profile::ProfileExists(profile))
         THROW_ERROR(cx, "Invalid profile specified");
-    char charname[24], file[_MAX_FNAME + MAX_PATH];
-    sprintf_s(file, sizeof(file), "%sd2bs.ini", Vars.szPath);
-    GetPrivateProfileString(profile, "character", "ERROR", charname, sizeof(charname), file);
+    wchar_t charname[24], file[_MAX_FNAME + MAX_PATH];
+    swprintf_s(file, sizeof(file), L"%sd2bs.ini", Vars.szPath);
+    GetPrivateProfileStringW(profile, L"character", L"ERROR", charname, sizeof(charname), file);

```

Comment:
Should switch sizeof(charname) to _countof(charname)

Reply:
Will fix these